llvm-bazel/.bazelrc

```diff
@@ -8,21 +8,48 @@
 ###############################################################################
 # Prevent invalid caching if input files are modified during a build.
 build --experimental_guard_against_concurrent_changes
-# Default to optimized builds
-# Override via: "-c dbg" or --compilation_mode=dbg
-build --compilation_mode=opt
 
 ###############################################################################
 # Options for "generic_clang" builds: these options should generally apply to
-# either clang or gcc and are curated based on need.
+# builds using a Clang-based compiler, and default to the `clang` executable on
+# the `PATH`. While these are provided for convenience and may serve as a
+# reference, it would be preferable for users to configure an explicit C++
+# toolchain instead of relying on `.bazelrc` files.
 ###############################################################################
 
+# Set the default compiler to the `clang` binary on the `PATH`.
+build:generic_clang --repo_env=CC=clang
```

Comment:
No need to set both `CXX` and `CC` from what I can see using Bazel, but let me know if I've missed something.

As for why setting it this way -- it makes it much easier for me to test that both Clang and GCC are working cleanly to handle it on the commandline as a config rather than relying on a particular environment.

Reply:
I guess if it works that's proof enough for me :-D

Makes sense. Maybe we should adopt this in IREE

Reply:
The disadvantage here is that someone who has clang at /my/weird/path and has their CC set to that is going to have to set it again with an explicit repo_env call, which is maybe too bad. Probably not a big deal though.

Reply:
Yeah, but if they are doing that they can just pass no `--config` at all and it should work.

Lemme know if you'd like me to factor the configs into one that sets `CC` and another that adds flags so that people can set `CC` themselves if they want and then still pick up our suggested flags. As I suspect these are mostly for the use of those developing the llvm-bazel project, it didn't seem worth doing yet, but I'm happy to add it.

Reply:
Nah this is fine for now